diff --git a/libraries/lzma/C/7zCrc.c b/libraries/lzma/C/7zCrc.c
index f186324..f3ae7be 100644
--- a/libraries/lzma/C/7zCrc.c
+++ b/libraries/lzma/C/7zCrc.c
@@ -83,12 +83,6 @@ UInt32 MY_FAST_CALL CrcUpdateT1(UInt32 v, const void *data, size_t size, const U
     #endif
   #elif (defined(__clang__) && (__clang_major__ >= 3)) \
      || (defined(__GNUC__) && (__GNUC__ > 4))
-      #if !defined(__ARM_FEATURE_CRC32)
-        #define __ARM_FEATURE_CRC32 1
-          #if (!defined(__clang__) || (__clang_major__ > 3)) // fix these numbers
-            #define ATTRIB_CRC __attribute__((__target__("arch=armv8-a+crc")))
-          #endif
-      #endif
       #if defined(__ARM_FEATURE_CRC32)
         #define USE_ARM64_CRC
         #include <arm_acle.h>
