diff --git a/meson.build b/meson.build
index 8db4dd4..e5e7549 100644
--- a/meson.build
+++ b/meson.build
@@ -396,7 +396,7 @@ endif
 if not speexdsp_dep.found() or not is_system_speexdsp_reliable
     speexdsp_dep = subproject(
         'speexdsp',
-        default_options: default_wrap_options,
+        default_options: default_wrap_options + ['simd=false']
     ).get_variable('speexdsp_dep')
     speexdsp_summary_msg = 'built-in'
 endif
