diff --git a/trunk/src/CMakeLists.txt b/trunk/src/CMakeLists.txt
index 5be7af8..8a260bb 100644
--- a/trunk/src/CMakeLists.txt
+++ b/trunk/src/CMakeLists.txt
@@ -165,7 +165,13 @@ source_group("Header Files" FILES ${BOMBERMAAAN_HEADERS} ${BOMBERMAAAN_DX_HEADER
 source_group("Source Files" FILES ${BOMBERMAAAN_SOURCES} ${BOMBERMAAAN_DX_SOURCES} ${BOMBERMAAAN_NET_SOURCES} ${WINREPLACE_SOURCES})
 
 find_package(SDL2 REQUIRED)
-find_package(SDL2_mixer REQUIRED)
+find_package(SDL2_mixer QUIET)
+if(NOT SDL2_mixer_FOUND)
+    find_library(SDL2_MIXER_LIBRARIES NAMES SDL2_mixer)
+    if(NOT SDL2_MIXER_LIBRARIES)
+        MESSAGE(FATAL_ERROR "SDL2_mixer libraries missing !")
+    endif()
+endif()
 
 if(USE_ALLEGRO)
   find_package(Allegro4 REQUIRED)
@@ -182,7 +188,7 @@ if(CRASH_REPORT)
 endif()
 
 set(BOMBERMAAAN_INCLUDE_DIRS
-  ${SDL2_INCLUDE_DIR}
+  ${SDL2_INCLUDE_DIR} ${SDL2_INCLUDE_DIRS}
   ${SDL2_MIXER_INCLUDE_DIR}
 )
 
@@ -207,8 +213,8 @@ include_directories(
 )
 
 set(BOMBERMAAAN_LIBRARIES
-  SDL2::SDL2
-  SDL2_mixer::SDL2_mixer
+  $<IF:$<TARGET_EXISTS:SDL2::SDL2>,SDL2::SDL2,${SDL2_LIBRARIES}>
+  $<IF:$<TARGET_EXISTS:SDL2_mixer::SDL2_mixer>,SDL2_mixer::SDL2_mixer,${SDL2_MIXER_LIBRARIES}>
   tinyxml
 )
 
