--- a/Makefile	2013-02-19 14:54:03.000000000 +0000
+++ b/Makefile	2017-01-29 04:51:05.264262542 +0000
@@ -512,7 +512,7 @@
 # SDLバージョンでの設定
 
 CFLAGS += -Isrc/FUNIX -Isrc/SDL `$(SDL_CONFIG) --cflags`
-LIBS   +=                       `$(SDL_CONFIG) --libs`
+LIBS   += -lm                   `$(SDL_CONFIG) --libs`
 
 CFLAGS += -DQUASI88_SDL
 
@@ -937,7 +937,7 @@
 all:		$(OBJDIRS) $(PROGRAM)
 
 $(OBJDIRS):
-		-mkdir $@
+		-mkdir -p $@
 
 $(PROGRAM):	$(OBJECTS)
 		$(LD) $(OBJECTS) $(LIBS) -o $(PROGRAM) 
--- a/src/fmgen/psg.cpp	2017-02-04 19:37:09.236638871 +0000
+++ b/src/fmgen/psg.cpp	2017-02-04 19:37:13.368626054 +0000
@@ -118,7 +118,7 @@
 		2,2, 2,0, 2,1, 2,3, 1,1, 1,3, 1,2, 1,0,
 	};
 	static uint8 table2[4] = {  0,  0, 31, 31 };
-	static uint8 table3[4] = {  0,  1, -1,  0 };
+	static uint8 table3[4] = {  0,  1, (uint8)-1,  0 };
 
 	uint* ptr = enveloptable[0];
