diff -u quasi88-0.6.4_org/Makefile quasi88-0.6.4/Makefile
--- quasi88-0.6.4_org/Makefile	2013-02-19 14:54:03.000000000 +0000
+++ quasi88-0.6.4/Makefile	2017-01-29 04:51:05.264262542 +0000
@@ -16,8 +16,8 @@
 #	GTK版の場合は、「GTK_VERSION」以外の行をコメントアウトしてください。
 #	( GTK版は実験中です。GTK版でサウンドを鳴らすには、SDL が必要です )
 
-X11_VERSION	= 1
-# SDL_VERSION	= 1
+# X11_VERSION	= 1
+SDL_VERSION	= 1
 # GTK_VERSION	= 1
 
 
@@ -32,7 +32,7 @@
 #	ディレクトリが使用されます。
 #	 ~/ は、QUASI88の起動時にホームディレクトリ（$HOME）に展開されます。
 
-ROMDIR	= ~/quasi88/rom/
+ROMDIR	= /home/pi/RetroPie/BIOS/quasi88
 
 
 # DISK用ディレクトリを設定します
@@ -42,7 +42,7 @@
 #	ディレクトリが使用されます。
 #	 ~/ は、QUASI88の起動時にホームディレクトリ（$HOME）に展開されます。
 
-DISKDIR	= ~/quasi88/disk/
+DISKDIR	= /home/pi/RetroPie/roms/pc88
 
 
 # TAPE用ディレクトリを設定します
@@ -51,7 +51,7 @@
 #	ディレクトリが使用されます。
 #	 ~/ は、QUASI88の起動時にホームディレクトリ（$HOME）に展開されます。
 
-TAPEDIR	= ~/quasi88/tape/
+TAPEDIR	= /home/pi/RetroPie/roms/pc88
 
 
 # (X11)
@@ -188,9 +188,9 @@
 #	動作確認が取れているのは、FreeBSD と Linux のみです
 
 #---------------------- FreeBSD
-ARCH = freebsd
+# ARCH = freebsd
 #---------------------- Linux
-# ARCH = linux
+ARCH = linux
 #---------------------- NetBSD
 # ARCH = netbsd
 #---------------------- OpenBSD
@@ -225,7 +225,7 @@
 # SOUND_ALSA		= 1
 # SOUND_ARTS_TEIRA	= 1
 # SOUND_ARTS_SMOTEK	= 1
-# SOUND_SDL		= 1
+SOUND_SDL		= 1
 # SOUND_WAVEOUT		= 1
 
 
@@ -234,7 +234,7 @@
 # サウンド出力を使用したい場合は、以下のコメントアウトを外して下さい。
 #	動作に必要なリソースが少なめとなります。
 
-# USE_OLD_MAME_SOUND	= 1
+USE_OLD_MAME_SOUND	= 1
 
 
 
@@ -245,7 +245,7 @@
 # cisc氏作の、fmgen (FM Sound Generator) を組み込まない場合、以下の行を
 # コメントアウトして下さい。
 
-USE_FMGEN	= 1
+# USE_FMGEN	= 1
 
 
 # 注意！
@@ -367,7 +367,7 @@
 # インストール先ディレクトリの設定
 #
 
-BINDIR = /usr/local/bin
+BINDIR = /opt/retropie/emulators/quasi88
 
 
 
@@ -512,7 +512,7 @@
 # SDLバージョンでの設定
 
 CFLAGS += -Isrc/FUNIX -Isrc/SDL `$(SDL_CONFIG) --cflags`
-LIBS   +=                       `$(SDL_CONFIG) --libs`
+LIBS   += -lm                   `$(SDL_CONFIG) --libs`
 
 CFLAGS += -DQUASI88_SDL
 
