diff --git a/ui/sdl/sdldisplay.c b/ui/sdl/sdldisplay.c
index e4d640b..fafe0ac 100644
--- a/ui/sdl/sdldisplay.c
+++ b/ui/sdl/sdldisplay.c
@@ -249,11 +249,13 @@ uidisplay_init( int width, int height )
     return 0;
   }
 
-  for( i=0; modes[i]; ++i ); /* count modes */
-  if( settings_current.sdl_fullscreen_mode ) {
-    if( sscanf( settings_current.sdl_fullscreen_mode, " %dx%d", &mw, &mh ) != 2 ) {
-      if( sscanf( settings_current.sdl_fullscreen_mode, " %d", &mn ) == 1 && mn <= i ) {
-        mw = modes[mn - 1]->w; mh = modes[mn - 1]->h;
+  if ( ! no_modes ) {
+    for( i=0; modes[i]; ++i ); /* count modes */
+    if( settings_current.sdl_fullscreen_mode ) {
+      if( sscanf( settings_current.sdl_fullscreen_mode, " %dx%d", &mw, &mh ) != 2 ) {
+        if( sscanf( settings_current.sdl_fullscreen_mode, " %d", &mn ) == 1 && mn <= i ) {
+          mw = modes[mn - 1]->w; mh = modes[mn - 1]->h;
+        }
       }
     }
   }
