diff --git a/GLideN64/src/Combiner.cpp b/GLideN64/src/Combiner.cpp
index f724cc0..0141f6c 100644
--- a/GLideN64/src/Combiner.cpp
+++ b/GLideN64/src/Combiner.cpp
@@ -295,7 +295,7 @@ void CombinerInfo::setCombine(u64 _mux )
 	} else {
 		m_pCurrent = Combiner_Compile(key);
 		m_pCurrent->update(true);
-		m_combiners[m_pCurrent->getKey()] = m_pCurrent;
+		m_combiners.emplace(m_pCurrent->getKey(), m_pCurrent);
 	}
 	m_bChanged = true;
 }
