diff --git a/GLideN64/src/Graphics/OpenGLContext/opengl_GLInfo.cpp b/GLideN64/src/Graphics/OpenGLContext/opengl_GLInfo.cpp
index 4f16a19..98facc5 100644
--- a/GLideN64/src/Graphics/OpenGLContext/opengl_GLInfo.cpp
+++ b/GLideN64/src/Graphics/OpenGLContext/opengl_GLInfo.cpp
@@ -154,7 +154,7 @@ void GLInfo::init() {
 	texture_barrierNV = Utils::isExtensionSupported(*this, "GL_NV_texture_barrier");
 
 	ext_fetch = Utils::isExtensionSupported(*this, "GL_EXT_shader_framebuffer_fetch") && !isGLES2 && (!isGLESX || ext_draw_buffers_indexed) && !imageTextures;
-	eglImage = (Utils::isEGLExtensionSupported("EGL_KHR_image_base") || Utils::isEGLExtensionSupported("EGL_KHR_image"));
+	eglImage = true; //(Utils::isEGLExtensionSupported("EGL_KHR_image_base") || Utils::isEGLExtensionSupported("EGL_KHR_image"));
 
 #ifdef OS_ANDROID
 	eglImage = eglImage &&
