--- a/configure	2018-08-22 14:35:59.974063916 +0000
+++ b/configure	2018-08-22 14:36:36.379862346 +0000
@@ -3274,7 +3274,7 @@
 			# We prefer SDL2 on the Raspberry Pi: acceleration now depends on it
 			# since SDL2 manages dispmanx/GLES2 very well internally.
 			# SDL1 is bit-rotten on this platform.
-			_sdlconfig=sdl2-config
+			_sdlconfig=sdl-config
 			# OpenGL ES support is mature enough as to be the best option on
 			# the Raspberry Pi, so it's enabled by default.
 			# The Raspberry Pi always supports OpenGL ES 2.0 contexts, thus we
