diff --git a/CMakeLists.txt b/CMakeLists.txt
index 299681c..41b971e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -405,6 +405,8 @@ if(USE_FFMPEG)
           set(PLATFORM_ARCH "linux/armv7")
         elseif(ARM)
           set(PLATFORM_ARCH "linux/arm")
+        elseif(ARM64)
+          set(PLATFORM_ARCH "linux/arm64")
         elseif(MIPS)
           set(PLATFORM_ARCH "linux/mips32")
         elseif(CMAKE_SIZEOF_VOID_P EQUAL 8)
