diff --git a/cmake/Toolchains/raspberry.armv6.cmake b/cmake/Toolchains/raspberry.armv6.cmake
index 8b05f08..602e5f5 100644
--- a/cmake/Toolchains/raspberry.armv6.cmake
+++ b/cmake/Toolchains/raspberry.armv6.cmake
@@ -8,20 +8,16 @@ include_directories(SYSTEM
   /opt/vc/include/interface/vmcx_host/linux
 )
 
-link_directories(
-  /opt/vc/lib
-)
-
 add_definitions(
   -DPPSSPP_PLATFORM_RPI=1
 )
 
 set(ARCH_FLAGS "-mfpu=vfp -march=armv6j -mfloat-abi=hard")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAGS}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_FLAGS}")
-set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${ARCH_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
+set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
 
-set(OPENGL_LIBRARIES GLESv2 bcm_host)
+set(OPENGL_LIBRARIES /opt/vc/lib/libGLESv2.so /opt/vc/lib/libbcm_host.so)
 set(USING_GLES2 ON)
 set(USING_FBDEV ON)
 
diff --git a/cmake/Toolchains/raspberry.armv7.cmake b/cmake/Toolchains/raspberry.armv7.cmake
index 7617829..9d900db 100644
--- a/cmake/Toolchains/raspberry.armv7.cmake
+++ b/cmake/Toolchains/raspberry.armv7.cmake
@@ -8,21 +8,17 @@ include_directories(SYSTEM
   /opt/vc/include/interface/vmcx_host/linux
 )
 
-link_directories(
-  /opt/vc/lib
-)
-
 add_definitions(
   -DPPSSPP_PLATFORM_RPI=1
   -U__GCC_HAVE_SYNC_COMPARE_AND_SWAP_2
 )
 
 set(ARCH_FLAGS "-mcpu=cortex-a7 -mfpu=neon -mfloat-abi=hard")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAGS}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_FLAGS}")
-set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${ARCH_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAGS}"  CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
+set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
 
-set(OPENGL_LIBRARIES GLESv2 bcm_host)
+set(OPENGL_LIBRARIES /opt/vc/lib/libGLESv2.so /opt/vc/lib/libbcm_host.so)
 set(USING_GLES2 ON)
 set(USING_FBDEV ON)
 set(ARMV7 ON)
