diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5193b756a2..4239305158 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,7 @@ endif()
 
 if(NOT ANDROID AND NOT IOS)
 	if(ARM OR SIMULATOR)
-		set(USING_EGL ON)
+		set(USING_EGL OFF)
 	endif()
 endif()
 
@@ -90,7 +90,7 @@ endif()
 
 # We only support Vulkan on Unix, Android and Windows.
 if(ANDROID OR WIN32 OR (UNIX AND NOT APPLE AND NOT ARM_NO_VULKAN))
-	set(VULKAN ON)
+	set(VULKAN OFF)
 endif()
 
 if(NOT DEFINED HEADLESS)
