diff --git a/Common/Render/Text/draw_text_sdl.h b/Common/Render/Text/draw_text_sdl.h
index 87c543a..810392f 100644
--- a/Common/Render/Text/draw_text_sdl.h
+++ b/Common/Render/Text/draw_text_sdl.h
@@ -10,7 +10,7 @@
 #endif
 
 // SDL2_ttf's TTF_Font is a typedef of _TTF_Font.
-struct _TTF_Font;
+#include <SDL2/SDL_ttf.h>
 
 class TextDrawerSDL : public TextDrawer {
 public:
@@ -33,12 +33,12 @@ class TextDrawerSDL : public TextDrawer {
 	bool FindFallbackFonts(uint32_t missingGlyph, int ptSize);
 
 	uint32_t fontHash_;
-	std::map<uint32_t, _TTF_Font *> fontMap_;
+	std::map<uint32_t, TTF_Font *> fontMap_;
 
 	std::map<CacheKey, std::unique_ptr<TextStringEntry>> cache_;
 	std::map<CacheKey, std::unique_ptr<TextMeasureEntry>> sizeCache_;
 
-	std::vector<_TTF_Font *> fallbackFonts_;
+	std::vector<TTF_Font *> fallbackFonts_;
 	std::vector<std::pair<std::string, int>> fallbackFontPaths_; // path and font face index
 
 #if defined(USE_SDL2_TTF_FONTCONFIG)
