diff --git a/retroarch.c b/retroarch.c
index 9850c21..0b6da5a 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -2598,7 +2598,8 @@ int generic_menu_entry_action(
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         // disable search action
+         // menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
