diff --git a/retroarch.c b/retroarch.c
index 0158f5c..a79bc98 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -988,6 +988,10 @@ static bool menu_widgets_fast_forward        = false;
 static bool menu_widgets_rewinding           = false;
 #endif
 
+/* number of frames required to trigger the hotkey */
+#define HOTKEY_DELAY 5
+static unsigned hotkey_counter = 0;
+
 bool menu_widgets_ready(void)
 {
 #ifdef HAVE_MENU_WIDGETS
@@ -13835,9 +13839,16 @@ static void input_keys_pressed(input_bits_t *p_new_state)
                   current_input_data, joypad_info,
                   &binds, port,
                   RETRO_DEVICE_JOYPAD, 0, RARCH_ENABLE_HOTKEY))
-            input_driver_block_libretro_input = true;
-         else
-            input_driver_block_hotkey         = true;
+            {
+               if (hotkey_counter < HOTKEY_DELAY)
+                  hotkey_counter++;
+               if (hotkey_counter == HOTKEY_DELAY)
+                   input_driver_block_libretro_input = true;
+            }
+            else {
+               hotkey_counter = 0;
+               input_driver_block_hotkey = true;
+            }
       }
 
       if (binds[RARCH_GAME_FOCUS_TOGGLE].valid)
