diff --git a/retroarch.c b/retroarch.c
index be36eaf141..dcccdaff47 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -1191,6 +1191,10 @@ static char current_valid_extensions[1024]                      = {0};
 static char error_string[255]                                   = {0};
 static char cached_video_driver[32]                             = {0};
 
+/* number of frames required to trigger the hotkey */
+#define HOTKEY_DELAY 5
+static unsigned hotkey_counter = 0;
+
 /* PATHS */
 struct rarch_dir_list
 {
@@ -18086,9 +18090,16 @@ static void input_keys_pressed(input_bits_t *p_new_state,
                current_input_data, joypad_info,
                &binds, port,
                RETRO_DEVICE_JOYPAD, 0, RARCH_ENABLE_HOTKEY))
-         input_driver_block_libretro_input = true;
-      else
-         input_driver_block_hotkey         = true;
+            {
+               if (hotkey_counter < HOTKEY_DELAY)
+                  hotkey_counter++;
+               if (hotkey_counter == HOTKEY_DELAY)
+                   input_driver_block_libretro_input = true;
+            }
+            else {
+               hotkey_counter = 0;
+               input_driver_block_hotkey = true;
+            }
    }
 
    if (binds[RARCH_GAME_FOCUS_TOGGLE].valid)
