diff --git a/menu/menu_driver.c b/menu/menu_driver.c
index 39e8e23527..4f673a2554 100644
--- a/menu/menu_driver.c
+++ b/menu/menu_driver.c
@@ -415,7 +415,8 @@ int generic_menu_entry_action(
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         // Disable search action
+         // menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
