diff --git a/menu/menu_driver.c b/menu/menu_driver.c
index cf3a8c3..aae5231 100644
--- a/menu/menu_driver.c
+++ b/menu/menu_driver.c
@@ -785,7 +785,7 @@ int menu_entry_action(menu_entry_t *entry,
                   entry->label, entry->type, i);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         // menu_input_dialog_start_search();
          break;
 
       case MENU_ACTION_SCAN:
