diff --git a/menu/widgets/menu_entry.c b/menu/widgets/menu_entry.c
index ea5be0c4cc..846f662c8f 100644
--- a/menu/widgets/menu_entry.c
+++ b/menu/widgets/menu_entry.c
@@ -474,7 +474,7 @@ int menu_entry_action(menu_entry_t *entry, unsigned i, enum menu_action action)
                   entry->label, entry->type, i);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         //menu_input_dialog_start_search();
          break;
 
       case MENU_ACTION_SCAN:
