diff --git a/input/drivers_joypad/udev_joypad.c b/input/drivers_joypad/udev_joypad.c
index c420d2b18b..27cd162dfa 100644
--- a/input/drivers_joypad/udev_joypad.c
+++ b/input/drivers_joypad/udev_joypad.c
@@ -599,8 +599,8 @@ static bool udev_joypad_init(void *data)
 
    /* Sort the udev entries by devnode name so that they are
     * created in the proper order */
-   qsort(sorted, sorted_count,
-         sizeof(struct joypad_udev_entry), sort_devnodes);
+   //qsort(sorted, sorted_count,
+   //      sizeof(struct joypad_udev_entry), sort_devnodes);
 
    for (i = 0; i < sorted_count; i++)
    {
