diff --git a/configuration.c b/configuration.c
index 46995c0aee..26951db746 100644
--- a/configuration.c
+++ b/configuration.c
@@ -1210,6 +1210,8 @@ static struct config_path_setting *populate_settings_path(settings_t *settings,
          settings->paths.path_core_options, false, NULL, true);
    SETTING_PATH("libretro_info_path",
          settings->paths.path_libretro_info, false, NULL, true);
+   SETTING_PATH("video_shader",
+          settings->paths.path_shader, false, NULL, true);
    SETTING_PATH("content_database_path",
          settings->paths.path_content_database, false, NULL, true);
    SETTING_PATH("cheat_database_path",
@@ -2270,6 +2272,7 @@ void config_set_defaults(void *data)
    *settings->paths.path_content_image_history   = '\0';
    *settings->paths.path_content_video_history   = '\0';
    *settings->paths.path_cheat_settings    = '\0';
+   *settings->paths.path_shader   = '\0';
 #ifndef IOS
    *settings->arrays.bundle_assets_src = '\0';
    *settings->arrays.bundle_assets_dst = '\0';
@@ -4170,6 +4173,10 @@ bool config_save_overrides(enum override_type type, void *data)

       for (i = 0; i < (unsigned)path_settings_size; i++)
       {
+         /* blacklist video_shader, better handled by shader presets*/
+         if (string_is_equal(path_settings[i].ident, "video_shader"))
+            continue;
+
          if (!string_is_equal(path_settings[i].ptr, path_overrides[i].ptr))
             config_set_path(conf, path_overrides[i].ident,
                   path_overrides[i].ptr);
diff --git a/configuration.h b/configuration.h
index 75b2eb4db0..b07108a2bc 100644
--- a/configuration.h
+++ b/configuration.h
@@ -706,6 +706,7 @@ typedef struct settings
       char path_libretro_info[PATH_MAX_LENGTH];
       char path_cheat_settings[PATH_MAX_LENGTH];
       char path_font[PATH_MAX_LENGTH];
+      char path_shader[PATH_MAX_LENGTH];
       char path_rgui_theme_preset[PATH_MAX_LENGTH];

       char directory_audio_filter[PATH_MAX_LENGTH];
diff --git a/retroarch.c b/retroarch.c
index c20ddae412..abcabd8494 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -27716,6 +27716,7 @@ static bool retroarch_load_shader_preset_internal(
 {
    unsigned i;
    char *shader_path = (char*)malloc(PATH_MAX_LENGTH);
+   settings_t *settings = configuration_settings;

    static enum rarch_shader_type types[] =
    {
@@ -27743,11 +27744,13 @@ static bool retroarch_load_shader_preset_internal(
          if (string_is_empty(special_name))
             break;

-         fill_pathname_join(shader_path, shader_directory, special_name, PATH_MAX_LENGTH);
-         strlcat(shader_path,
-               video_shader_get_preset_extension(types[i]),
-               PATH_MAX_LENGTH);
-      }
+         if (strcmp(special_name, "config")!=0) {
+            fill_pathname_join(shader_path, shader_directory, special_name, PATH_MAX_LENGTH);
+            strlcat(shader_path, video_shader_get_preset_extension(types[i]), PATH_MAX_LENGTH);
+         } else {
+             strlcpy(shader_path, settings->paths.path_shader, PATH_MAX_LENGTH);
+         }
+       }

       if (!path_is_valid(shader_path))
          continue;
@@ -27771,6 +27774,7 @@ static bool retroarch_load_shader_preset_internal(
  * Tries to load a supported core-, game-, folder-specific or global
  * shader preset from its respective location:
  *
+ * config:          preset from the configuration file, configured via 'video_shader'
  * global:          $CONFIG_DIR/global.$PRESET_EXT
  * core-specific:   $CONFIG_DIR/$CORE_NAME/$CORE_NAME.$PRESET_EXT
  * folder-specific: $CONFIG_DIR/$CORE_NAME/$FOLDER_NAME.$PRESET_EXT
@@ -27879,6 +27883,15 @@ static bool retroarch_load_shader_preset(void)
          RARCH_LOG("[Shaders]: global shader preset found.\n");
          break;
       }
+
+      ret = retroarch_load_shader_preset_internal(dirs[i], NULL,
+         "config");
+
+      if (ret)
+      {
+         RARCH_LOG("[Shaders]: configuration file shader preset found.\n");
+         break;
+      }
    }

 end:
