diff --git a/gfx/drivers_context/mali_fbdev_ctx.c b/gfx/drivers_context/mali_fbdev_ctx.c
index 1b7110d..5d8b9d3 100644
--- a/gfx/drivers_context/mali_fbdev_ctx.c
+++ b/gfx/drivers_context/mali_fbdev_ctx.c
@@ -44,7 +44,10 @@ typedef struct
    egl_ctx_data_t egl;
 #endif
 
-   struct mali_native_window native_window;
+   struct {
+      unsigned short width;
+      unsigned short height;
+   } native_window;
    bool resize;
    unsigned width, height;
 } mali_ctx_data_t;
