diff --git a/backends/platform/libretro/os.cpp b/backends/platform/libretro/os.cpp
index c37171e..eb4c222 100644
--- a/backends/platform/libretro/os.cpp
+++ b/backends/platform/libretro/os.cpp
@@ -1212,7 +1212,7 @@ class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
       {
          Common::Event ev;
          ev.type = Common::EVENT_QUIT;
-         _events.push_back(ev);
+         ((OSystem_RETRO*)g_system)->getEventManager()->pushEvent(ev);
       }
 };
 
